libs/wingc/src/type_check.rs

```diff
@@ -757,6 +757,11 @@ impl Subtype for Type {
 				// Nil is a subtype of Optional<T> for any T
 				true
 			}
+			(Self::Void, Self::Optional(_)) => {
```

Comment:
Can you please add a test?

Reply:
Added a test, let me know if it needs to be changed or in a different file.